### DISCUSSION TITLE
esp32-c3 wakeup pin not working

### DISCUSSION THREAD

**Discussion by puppet13th** [micropython/micropython#9482] [Hardware & Peripherals]
board lolin-c3-mini using generic esp32-c3-usb 1.19.1
example code:
```py
from machine import Pin,lightsleep
from time import ticks_ms,sleep_ms
import esp32

led = Pin(7,Pin.OUT,value=0)

for i in [0,1,2,3,4,5]:
    try:
        pin = Pin(i,Pin.IN)
        print(pin,pin.value())
        try:
            esp32.wake_on_ext0(pin = pin, level = esp32.WAKEUP_ANY_HIGH)
        except Exception as exc:
            print(i,'wake',exc)
        else:
            ticks = ticks_ms()
            lightsleep(10_000)
            if ticks_ms() - ticks < 10_000: # wakeup before timeout
                print(i,'wake')
                led.on()
                sleep_ms(1000)
                led.off()
                sleep_ms(200)
                for count in range(0,i): # blink led
                    led.on()
                    sleep_ms(200)
                    led.off()
                    sleep_ms(200)
    except Exception as exc:
        print(i,'input',exc)

led.on()

```
code output if "lightsleep(10_000)" comment out:
```py
MicroPython v1.19.1 on 2022-06-18; ESP32C3 module with ESP32C3
Type "help()" for more information.
>>> %run -c $EDITOR_CONTENT
Pin(0) 1
0 wake
Pin(1) 1
1 wake invalid pin
Pin(2) 1
2 wake
Pin(3) 1
3 wake invalid pin
Pin(4) 1
4 wake
Pin(5) 1
5 wake invalid pin
```
according to [esp32-c3 datasheet](https://www.espressif.com/sites/default/files/documentation/esp32-c3_datasheet_en.pdf)/[another reference](https://docs.espressif.com/projects/esp-idf/en/latest/esp32c3/api-reference/peripherals/gpio.html) pin 1,3,5 connected to VDD3P3_RTC and should be able to be used as wakeup pin.
the led only lit at the end of the code(esp32.wake_on_ext0 not working)

the code run correctly(led blinking according to pin number) on lolin-s3:
code output:
```
MicroPython v1.19.1 on 2022-09-25; LOLIN S3 with ESP32S3
Type "help()" for more information.
>>> %run -c $EDITOR_CONTENT
Pin(0) 1
0 wake
Pin(1) 1
1 wake
Pin(2) 1
2 wake
Pin(3) 1
3 wake
Pin(4) 1
4 wake
Pin(5) 1
5 wake
```

edit:
wiring:
v3v3 - r1 - 'pin0 - pin5'

**Comment by pbdaran** [comment]
@puppet13th Is the issue fixed?
I am trying to use GPIO1 of ESP32-C3 as wakeup pin. But unable to wakeup the device.

**Comment by ricksorensen** [comment]
See pending pull requests:
https://github.com/micropython/micropython/pull/8995 Fix ESP32-C3 deep/light sleep wake on GPIOs support
https://github.com/micropython/micropython/pull/9583 modmachine: Implemented GPIO wake-up for ESP32-C3

**Reply by pbdaran** [reply]
Got it. Thank you.
